Initial dev for #22

bradpowell opened the pull request:
Added View and Controller for ContactUs Page.  This is pretty basic so
there is no need for a model (I think).

Updated Header's ContactUs anchor to point to new page.

Let me know if I've made any MVC rookie mistakes.

Question: What's the best practice for updating the navbar2's list items to reflect which page is active?  Right now it's hardcoded to show Home as active.  Do we give each list item an ID and update it via the dom (with jquery) from the page being viewed?

bradpowell committed:
Initial dev for #22

Added View and Controller for ContactUs Page.  This is pretty basic so
there is no need for a model (I think).

Updated Header's ContactUs anchor to point to new page.

bradpowell committed:
Initial dev for #22

Added View and Controller for ContactUs Page.  This is pretty basic so
there is no need for a model (I think).

Updated Header's ContactUs anchor to point to new page.

mdryden commented on src/Website/Views/ContactUs/ContactUs.cshtml:
```diff
@@ -0,0 +1,24 @@
+﻿
+@{
+    ViewBag.Title = "Contact Us";
```
I think there might be a couple other files that use the ViewBag for the title, but I'd like to eliminate the use of the ViewBag wherever possible, so I actually added a "Title" property to the page base class in a prior commit.

Lets use:
`this.PageTitle = "Contact Us";`
instead.

Also, we should localize that string.

bradpowell replied:
Updated to use the PageTitle property and localized.

mdryden commented on src/Website/Views/ContactUs/ContactUs.cshtml:
```diff
@@ -0,0 +1,24 @@
+        <p>Email address goes here.</p>
+    </section>
+</article>
+
```
Since we're designing this site to be re-usable (in theory), should the about/contact info be stored in the database and configurable by the admins?

bradpowell replied:
Crossed my mind too, but it felt weird since it would either be it's own table with only ever one record in it or be stored in some non normalized catch all table.

Unless you have any strong objections I vote for leaving it as is for now.  Maybe when we dig into some of the other pages we'll think of something that could also be applied here.

mdryden commented on src/Website/Controllers/ContactUsController.cs:
```diff
@@ -0,0 +1,18 @@
+{
+    public class ContactUsController : Controller
+    {
+        [GET("ContactUs")]
```
My personal preference with URLs is for them to be all lower case, unless we're hitting something like say an URL which contains a username or something.  We should decide on a standard.

bradpowell replied:
I'm fine with going all lowercase if that's a standard, otherwise it strikes me as a bit odd to abandon camel or pascal casing which provide greater readability than something that is all of one casing type be it all upper or all lowercase.

mdryden commented on src/Website/Views/Shared/_Header.cshtml:
```diff
@@ -16,7 +16,7 @@
 				<li><a href="#">Login</a></li>
 				<li><a href="#about">Register</a></li>
-				<li><a href="#contact">Contact Us</a></li>
+				<li><a href="ContactUs">Contact Us</a></li>
```
To make sure this path works no matter where you are in the site, it should be relative.  (/contactus, most of the time).

MVC actually has a character that can be used to represent the installation folder structure (within IIS) to make sure it's relative within that:

href="~/ContactUs"

If we decide on that lowercase is the standard, we'll want to change this href too.

bradpowell replied:
I see that this was udpated at some point to use the UrlHelper.  I've updated that code to point at the about us controller and appropriate action.

bradpowell committed:
Updated from dev

Merge the latest code from Dev into branch

bradpowell committed:
Fixes in response to review of pull request

Initial fix for pull request comments.  Doesn't run however, will need
to debug that.

bradpowell committed:
Fixes in response to review of pull request

Initial fix for pull request comments.  Doesn't run however, will need
to debug that.

bradpowell committed:
Merge branch 'contact-us-#22' of https://github.com/mb-soft/BrewClub into contact-us-#22

bradpowell committed:
Implement suggested changes
